singolo/js/script.js

```diff
@@ -0,0 +1,253 @@
+    ) {
+      nav[0].classList.add("active");
+    } else {
+      nav[i].classList.remove("active");
+    }
+    if (
+      bodyChildren[2].offsetTop <= pageYOffset &&
+      pageYOffset <= bodyChildren[2].offsetTop + bodyChildren[2].scrollHeight
+    ) {
+      nav[1].classList.add("active");
+    } else {
+      nav[i].classList.remove("active");
+    }
+    if (
+      bodyChildren[3].offsetTop <= pageYOffset &&
+      pageYOffset < bodyChildren[3].offsetTop + bodyChildren[3].scrollHeight
+    ) {
+      nav[2].classList.add("active");
+    } else {
+      nav[i].classList.remove("active");
+    }
+  });
+  //CLOSURE THE BURGER MENU , HIGHLIGHTING THE MENU ON CLICK AND SMOOTH SCROLLING TO THE ELEMENT
+  nav[i].addEventListener("click", function (e) {
+    e.preventDefault();
+    remove.classList.remove("active");
+    nav[i].classList.add("active");
+    remove = nav[i];
+    burger.classList.remove("click");
+    header_content.classList.remove("open");
+    const blockID = nav[i].getAttribute("href").substr(1);
+    document.getElementById(blockID).scrollIntoView({
+      behavior: "smooth",
+      block: "start",
+    });
+  });
+}
+
+//HIGHLIGHTING THE BUTTONS FILTER ON CLICK
+
+for (let i = 0; i < button_filter.length; i++) {
+  button_filter[i].addEventListener("click", () => {
+    remove_active_button.classList.remove("active-button-filter");
+    button_filter[i].classList.add("active-button-filter");
+    remove_active_button = button_filter[i];
+  });
+}
+
+//SHUFFLING IMAGE ELEMENTS
+for (let s = 0; s < images.length; s++) {
+  all.addEventListener("click", () => {
+    images[s].style.backgroundImage =
+      "url(images/portfolio-images/portfolio-" + numb[s] + ".png)";
+  });
+}
+web.addEventListener("click", () => {
+  shuffle(arr);
+});
+for (let s = 0; s < images.length; s++) {
+  web.addEventListener("click", () => {
+    images[s].style.backgroundImage =
+      "url(images/portfolio-images/portfolio-" + arr[s] + ".png)";
+  });
+}
+graphic.addEventListener("click", () => {
+  shuffle(arr);
+});
+for (let s = 0; s < images.length; s++) {
+  graphic.addEventListener("click", () => {
+    images[s].style.backgroundImage =
+      "url(images/portfolio-images/portfolio-" + arr[s] + ".png)";
+  });
+}
+artwork.addEventListener("click", () => {
+  shuffle(arr);
+});
+for (let s = 0; s < images.length; s++) {
+  artwork.addEventListener("click", () => {
+    images[s].style.backgroundImage =
+      "url(images/portfolio-images/portfolio-" + arr[s] + ".png)";
+  });
+}
+
+function shuffle(array) {
+  let i = array.length,
+    temporaryValue,
+    random;
+  while (0 !== i) {
+    random = Math.floor(Math.random() * i);
+    i -= 1;
+    temporaryValue = array[i];
+    array[i] = array[random];
+    array[random] = temporaryValue;
+  }
+
+  return array;
+}
+let arr = [];
+for (let i = 1; i < 13; i++) {
+  numb.push(i);
+  arr.push(i);
+}
+shuffle(arr);
+
+//SLIDER
+let slider = document.getElementById("slider"),
```

Comment:
let's keep all global variables in one place.

Reply:
I agree